ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -670,6 +670,11 @@ assignToSolution(data::Solution& sol)
             std::string tmp = tracer.name + "F";
             sol.insert(tmp, UnitSystem::measure::identity, std::move(tracerConcentrations_[tracerIdx++]), data::TargetType::RESTART_SOLUTION);
         }
+        // We need put tracerConcentrations into a valid state.
+        // Otherwise next time we end up here outside of a restart write we will
+        // move invalidated data above (as it was moved away before and never
+        // reallocated)
```

Comment:
then please improve the interpretation for those not in the know that have to rely on code reading:

The inconsistency is between here and [eclgenericoutputblackoilmodule.cc#L858-L859](/OPM/opm-simulators/blob/5e1853b8e465668ddf57175de030b5242f680608/ebos/eclgenericoutputblackoilmodule.cc#L858-L859).
```
    if (!isRestart && (!schedule_.write_rst_file(reportStepNum) || substep))
        return;
```
The first time `assignToSolution` is called that condition was `false` because of `isRestart==false`, `schedule_.write_rst_file(0)==true`, and `subStep==false`. The next time that assignSolution was called it was `true` because of `isRestart==false`, `schedule_.write_rst_file(1)==false`, and `subStep==false`.

Reply:
Okay, on second thought I think I understand what you're getting at.  We get here on a report step (`substep == false`) that's not also a restart step (`schedule_.write_rst_file() == false`).  `IsRestart` is `true` only for the very first step of a restarted run (keyword `RESTART` in the `SOLUTION` section) so we don't need to worry about that here.  In other words, this is a non-restart report step in a run that uses something along the lines of
```
RPTRST
  BASIC=5 FREQ=6 /
```
(or any of the myriad other spellings that restrict restart output to a subset of the defined report steps (`DATES` or `TSTEP`)).  In that case, this is a quick fix that will solve the immediate problem.

My greater point still stands however, that we should never get here unless we're actually collecting values for restart file output.  I'll put fixing that on my list.